src/axom/sidre/core/Group.hpp

```diff
@@ -109,9 +109,25 @@ class MapCollection;
  * of the indexed group.  None of these methods is marked with "Child".
  *
  * A Group can optionally be created to hold items in a "list format". In
- * this format, any number of child Group or View items can be created with
- * empty strings for their names, and none of the methods that access
- * child items by name or path will return a valid pointer.
+ * this format, any number of child Group or View items can be created to
+ * be held and accessed like a list.  When using this format, the child items
+ * can be accessed only via iterators that iterate in the order that they
+ * were added to the parent Group.
+ *
+ * It is recommended but not required that the items held in the list format
+ * be created without names.  String names may be assigned to these items,
+ * but the names will not be useful for accessing them from their parent
+ * Group, and none of the methods that access child items by name or path will
+ * return a valid pointer.  Unnamed Groups to be held in the list format
+ * should be created using the method createUnnamedGroup, while unnamed
+ * Views should be created by passing an empty string as the path argument
+ * to any of the createView methods.
+ *
+ * For the list format ONLY, if the method creating a Group or View is
+ * passed a string with path syntax (such as "foo/bar/baz"), only the
+ * name after the final delimiter is assigned to the new item, and the
```

Comment:
Is there a reason for this behavior because, as a user, this is not what I would have expected to happen. I would have expected either all intermediate groups to be created (like `mkdir -p foo/bar/baz`) or error out that you cannot create a group with the special character `/` in it.

Reply:
It's worthy of discussion what should happen here; my choice might not be the best.  Creating the intermediate groups is not good, as it will orphan the object at the end of the path, since the path cannot be used from a list-format Group.  Thinking of it now, it may be better to have create give a warning and return a null pointer, as there's no good use case to provide a path string.  A null pointer rather than an error would be consistent with the design choice across Sidre to return a null that the user can check for failed object creation rather than crashing a run.

Reply:
Once the suggested change is implemented, created_group and found_group will be both be nullptr.  Test assertions can be added here to verify that.